draft-thomson-train-protocol.md

```diff
@@ -99,167 +105,322 @@ cannot assume that rate limit indications from one path apply to new paths.
 
 # Applicability
 
-This protocol only works for flows that use specific QUIC versions and only
-where those flows include packets using the long header {{Section 5.1 of
-!RFC8999}}.
+This protocol only works for flows that use the TRAIN packet ({{packet}}).
 
-The information provided to endpoints (or applications at those endpoints) is
-advisory.  The protocol requires that packets are modified as they transit a
-network element, which provides endpoints proof that the network element has the
-power to drop packets.  However, it does not prove that the rate limit that is
-indicated would either be successful; nor does it prove that a higher rate would
-not be successful.  Endpoints that receive this signal therefore need to treat
-the information as advisory.
+The protocol requires that packets are modified as they transit a
+network element, which provides endpoints strong evidence that the network
+element has the power to drop packets; though see {{security}} for potential
+limitations on this.
 
-As an advisory signal, network elements cannot assume that endpoints will
-respect the signal.  Though this might reduce the need for more active rate
-limiting, whether rate limit enforcement is necessary is a matter for network
-policy.
+The rate limit signal that this protocol carries is independent of congestion
+signals, limited to a single path and UDP packet flow, unidirectional, and
+strictly advisory.
+
+## Independent of Congestion Signals
+
+Rate limit signals are not a substitute for congestion feedback.  Congestion
+signals, such as acknowledgments, provide information on loss, delay, or ECN
+markings {{?ECN=RFC3168}} that indicate the real-time condition of a network
+path.  Congestion signals might indicate a throughput that is different from the
+signaled rate limit.
 
-The time and scope over which a rate limit applies is not specified.  Rate
-limits might change without being signaled.  The signaled limit can be assumed
-to apply to the flow of packets on the same UDP address tuple for the duration
-of that flow.  Rate limiting policies often apply on the level of a device or
-subscription, but endpoints cannot assume that this is the case.  A separate
-signal can be sent for each flow.
+Endpoints cannot assume that a signaled rate limit is achievable if congestion
+signals indicate otherwise.  That is, endpoints need to respect the send rate
+constraints that are set by a congestion controller.
+
+## On Path Signal
+
+Modifying a packet does not prove that the rate limit that is indicated would be
+achievable.  A signal that is sent for a specific flow might be enforced at a
+different scope.  For instance, limits might apply at a network subscription
+level, such that multiple flows receive the same signal.  Nor does a signal
+prove that a higher rate would not be successful.  Endpoints that receive this
+signal therefore need to treat the information as advisory.
```

Comment:
Not that we have to do in this PR, but do we want to point out that the congestion point might be somewhere else that the node that is advertising the rate limit?

Reply:
Good call, I added something.